Compile Error on Melodic

```
22:48:45 [ 74%] Building CXX object CMakeFiles/inflation_tests.dir/test/inflation_tests.cpp.o
22:48:50 /tmp/catkin_workspace/src/navigation/costmap_2d/test/inflation_tests.cpp: In function ‘void validatePointInflation(unsigned int, unsigned int, costmap_2d::Costmap2D*, costmap_2d::InflationLayer*, double)’:
22:48:50 /tmp/catkin_workspace/src/navigation/costmap_2d/test/inflation_tests.cpp:88:52: error: call of overloaded ‘abs(unsigned int)’ is ambiguous
22:48:50          unsigned int dx = abs(cell.x_ - cell.src_x_);
22:48:50                                                     ^
22:48:50 In file included from /usr/include/c++/7/cstdlib:75:0,
22:48:50                  from /usr/include/c++/7/ext/string_conversions.h:41,
22:48:50                  from /usr/include/c++/7/bits/basic_string.h:6349,
22:48:50                  from /usr/include/c++/7/string:52,
22:48:50                  from /usr/include/c++/7/stdexcept:39,
22:48:50                  from /usr/include/c++/7/array:39,
22:48:50                  from /usr/include/c++/7/tuple:39,
22:48:50                  from /usr/include/c++/7/bits/stl_map.h:63,
22:48:50                  from /usr/include/c++/7/map:61,
22:48:50                  from /tmp/catkin_workspace/src/navigation/costmap_2d/test/inflation_tests.cpp:34:
22:48:50 /usr/include/stdlib.h:837:12: note: candidate: int abs(int)
22:48:50  extern int abs (int __x) __THROW __attribute__ ((__const__)) __wur;
22:48:50             ^~~
22:48:50 In file included from /usr/include/c++/7/cstdlib:77:0,
22:48:50                  from /usr/include/c++/7/ext/string_conversions.h:41,
22:48:50                  from /usr/include/c++/7/bits/basic_string.h:6349,
22:48:50                  from /usr/include/c++/7/string:52,
22:48:50                  from /usr/include/c++/7/stdexcept:39,
22:48:50                  from /usr/include/c++/7/array:39,
22:48:50                  from /usr/include/c++/7/tuple:39,
22:48:50                  from /usr/include/c++/7/bits/stl_map.h:63,
22:48:50                  from /usr/include/c++/7/map:61,
22:48:50                  from /tmp/catkin_workspace/src/navigation/costmap_2d/test/inflation_tests.cpp:34:
22:48:50 /usr/include/c++/7/bits/std_abs.h:56:3: note: candidate: long int std::abs(long int)
22:48:50    abs(long __i) { return __builtin_labs(__i); }
22:48:50    ^~~
22:48:50 /usr/include/c++/7/bits/std_abs.h:61:3: note: candidate: long long int std::abs(long long int)
22:48:50    abs(long long __x) { return __builtin_llabs (__x); }
22:48:50    ^~~
22:48:50 /usr/include/c++/7/bits/std_abs.h:70:3: note: candidate: constexpr double std::abs(double)
22:48:50    abs(double __x)
22:48:50    ^~~
22:48:50 /usr/include/c++/7/bits/std_abs.h:74:3: note: candidate: constexpr float std::abs(float)
22:48:50    abs(float __x)
22:48:50    ^~~
22:48:50 /usr/include/c++/7/bits/std_abs.h:78:3: note: candidate: constexpr long double std::abs(long double)
22:48:50    abs(long double __x)
22:48:50    ^~~
22:48:50 /usr/include/c++/7/bits/std_abs.h:84:3: note: candidate: constexpr __int128 std::abs(__int128)
22:48:50    abs(__GLIBCXX_TYPE_INT_N_0 __x) { return __x >= 0 ? __x : -__x; }
22:48:50    ^~~
22:48:50 /usr/include/c++/7/bits/std_abs.h:102:3: note: candidate: constexpr __float128 std::abs(__float128)
22:48:50    abs(__float128 __x)
22:48:50    ^~~
22:48:50 /tmp/catkin_workspace/src/navigation/costmap_2d/test/inflation_tests.cpp:89:52: error: call of overloaded ‘abs(unsigned int)’ is ambiguous
22:48:50          unsigned int dy = abs(cell.y_ - cell.src_y_);
22:48:50                                                     ^
22:48:50 In file included from /usr/include/c++/7/cstdlib:75:0,
22:48:50                  from /usr/include/c++/7/ext/string_conversions.h:41,
22:48:50                  from /usr/include/c++/7/bits/basic_string.h:6349,
22:48:50                  from /usr/include/c++/7/string:52,
22:48:50                  from /usr/include/c++/7/stdexcept:39,
22:48:50                  from /usr/include/c++/7/array:39,
22:48:50                  from /usr/include/c++/7/tuple:39,
22:48:50                  from /usr/include/c++/7/bits/stl_map.h:63,
22:48:50                  from /usr/include/c++/7/map:61,
22:48:50                  from /tmp/catkin_workspace/src/navigation/costmap_2d/test/inflation_tests.cpp:34:
22:48:50 /usr/include/stdlib.h:837:12: note: candidate: int abs(int)
22:48:50  extern int abs (int __x) __THROW __attribute__ ((__const__)) __wur;
22:48:50             ^~~
22:48:50 In file included from /usr/include/c++/7/cstdlib:77:0,
22:48:50                  from /usr/include/c++/7/ext/string_conversions.h:41,
22:48:50                  from /usr/include/c++/7/bits/basic_string.h:6349,
22:48:50                  from /usr/include/c++/7/string:52,
22:48:50                  from /usr/include/c++/7/stdexcept:39,
22:48:50                  from /usr/include/c++/7/array:39,
22:48:50                  from /usr/include/c++/7/tuple:39,
22:48:50                  from /usr/include/c++/7/bits/stl_map.h:63,
22:48:50                  from /usr/include/c++/7/map:61,
22:48:50                  from /tmp/catkin_workspace/src/navigation/costmap_2d/test/inflation_tests.cpp:34:
22:48:50 /usr/include/c++/7/bits/std_abs.h:56:3: note: candidate: long int std::abs(long int)
22:48:50    abs(long __i) { return __builtin_labs(__i); }
22:48:50    ^~~
22:48:50 /usr/include/c++/7/bits/std_abs.h:61:3: note: candidate: long long int std::abs(long long int)
22:48:50    abs(long long __x) { return __builtin_llabs (__x); }
22:48:50    ^~~
22:48:50 /usr/include/c++/7/bits/std_abs.h:70:3: note: candidate: constexpr double std::abs(double)
22:48:50    abs(double __x)
22:48:50    ^~~
22:48:50 /usr/include/c++/7/bits/std_abs.h:74:3: note: candidate: constexpr float std::abs(float)
22:48:50    abs(float __x)
22:48:50    ^~~
22:48:50 /usr/include/c++/7/bits/std_abs.h:78:3: note: candidate: constexpr long double std::abs(long double)
22:48:50    abs(long double __x)
22:48:50    ^~~
22:48:50 /usr/include/c++/7/bits/std_abs.h:84:3: note: candidate: constexpr __int128 std::abs(__int128)
22:48:50    abs(__GLIBCXX_TYPE_INT_N_0 __x) { return __x >= 0 ? __x : -__x; }
22:48:50    ^~~
22:48:50 /usr/include/c++/7/bits/std_abs.h:102:3: note: candidate: constexpr __float128 std::abs(__float128)
22:48:50    abs(__float128 __x)
22:48:50    ^~~
22:48:51 make[3]: *** [CMakeFiles/inflation_tests.dir/test/inflation_tests.cpp.o] Error 1
22:48:51 CMakeFiles/inflation_tests.dir/build.make:62: recipe for target 'CMakeFiles/inflation_tests.dir/test/inflation_tests.cpp.o' failed
22:48:51 CMakeFiles/Makefile2:4681: recipe for target 'CMakeFiles/inflation_tests.dir/all' failed
22:48:51 make[2]: *** [CMakeFiles/inflation_tests.dir/all] Error 2
22:48:51 CMakeFiles/Makefile2:3039: recipe for target 'CMakeFiles/tests.dir/rule' failed
22:48:51 make[1]: *** [CMakeFiles/tests.dir/rule] Error 2
22:48:51 Makefile:1189: recipe for target 'tests' failed
22:48:51 make: *** [tests] Error 2
22:48:51 <== Failed to process package 'costmap_2d': 
22:48:51   Command '['/tmp/catkin_workspace/devel_isolated/voxel_grid/env.sh', 'make', '-j1', 'tests']' returned non-zero exit status 2
22:48:51 
22:48:51 Reproduce this error by running:
22:48:51 ==> cd /tmp/catkin_workspace/build_isolated/costmap_2d && /tmp/catkin_workspace/devel_isolated/voxel_grid/env.sh make -j1 tests
22:48:51 
22:48:51 Command failed, exiting.
```

--- a/amcl/src/amcl_node.cpp
+++ b/amcl/src/amcl_node.cpp
@@ -24,6 +24,7 @@
 #include <vector>
 #include <map>
 #include <cmath>
+#include <memory>
 
 #include <boost/bind.hpp>
 #include <boost/thread/mutex.hpp>

--- a/costmap_2d/include/costmap_2d/costmap_2d_ros.h
+++ b/costmap_2d/include/costmap_2d/costmap_2d_ros.h
@@ -46,7 +46,7 @@
 #include <geometry_msgs/Polygon.h>
 #include <geometry_msgs/PolygonStamped.h>
 #include <dynamic_reconfigure/server.h>
-#include <pluginlib/class_loader.h>
+#include <pluginlib/class_loader.hpp>
 
 class SuperValue : public XmlRpc::XmlRpcValue
 {

--- a/costmap_2d/test/footprint_tests.cpp
+++ b/costmap_2d/test/footprint_tests.cpp
@@ -170,6 +170,7 @@ int main(int argc, char** argv)
   base_rel_map.child_frame_id_ = "/base_link";
   base_rel_map.frame_id_ = "/map";
   base_rel_map.stamp_ = ros::Time::now();
+  base_rel_map.setIdentity();
   tf_->setTransform( base_rel_map );
 
   testing::InitGoogleTest(&argc, argv);

--- a/costmap_2d/test/inflation_tests.cpp
+++ b/costmap_2d/test/inflation_tests.cpp
@@ -32,6 +32,7 @@
  * Test harness for InflationLayer for Costmap2D
  */
 #include <map>
+#include <cmath>
 
 #include <costmap_2d/costmap_2d.h>
 #include <costmap_2d/layered_costmap.h>
@@ -85,8 +86,8 @@ void validatePointInflation(unsigned int mx, unsigned int my, Costmap2D* costmap
       if (!seen[cell.index_])
       {
         seen[cell.index_] = true;
-        unsigned int dx = abs(cell.x_ - cell.src_x_);
-        unsigned int dy = abs(cell.y_ - cell.src_y_);
+        unsigned int dx = (cell.x_ > cell.src_x_) ? cell.x_ - cell.src_x_ : cell.src_x_ - cell.x_;
+        unsigned int dy = (cell.y_ > cell.src_y_) ? cell.y_ - cell.src_y_ : cell.src_y_ - cell.y_;
         double dist = hypot(dx, dy);
 
         unsigned char expected_cost = ilayer->computeCost(dist);

--- a/move_base/include/move_base/move_base.h
+++ b/move_base/include/move_base/move_base.h
@@ -53,7 +53,7 @@
 #include <costmap_2d/costmap_2d.h>
 #include <nav_msgs/GetPlan.h>
 
-#include <pluginlib/class_loader.h>
+#include <pluginlib/class_loader.hpp>
 #include <std_srvs/Empty.h>
 
 #include <dynamic_reconfigure/server.h>

